File: src/main/java/org/jukeboxmc/event/entity/EntityEvent.java
package org.jukeboxmc.event.entity;

import org.jukeboxmc.entity.Entity;
import org.jukeboxmc.event.Event;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public abstract class EntityEvent extends Event {

    private Entity entity;

    public EntityEvent( Entity entity ) {
        this.entity = entity;
    }

    public Entity getEntity() {
        return this.entity;
    }

    public void setEntity( Entity entity ) {
        this.entity = entity;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/entity/EntityEvent.kt
package org.jukeboxmc.api.event.entity

import org.jukeboxmc.api.entity.Entity
import org.jukeboxmc.api.event.Event

open class EntityEvent(
    private val entity: Entity
) : Event() {

    open fun getEntity(): Entity {
        return this.entity
    }
}